component/loki/source/http/http.go

```diff
@@ -0,0 +1,170 @@
+package http
+
+import (
+	"context"
+	"reflect"
+	"sync"
+
+	"github.com/grafana/agent/pkg/util"
+
+	"github.com/go-kit/log/level"
+
+	"github.com/efficientgo/core/errors"
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/common/loki"
+	"github.com/grafana/agent/component/common/relabel"
+	"github.com/grafana/agent/component/loki/source/http/internal/lokipush"
+	"github.com/prometheus/common/model"
+	"github.com/weaveworks/common/logging"
+	"github.com/weaveworks/common/server"
+)
+
+// TODO: this component also supports GRPC, so we may want to call it `loki.source.push_api` or something else.
+const componentName = "loki.source.http"
```

Comment:
Good point, let's find a better name for the component.

My 2c, in order of preference:
* loki.source.api
* loki.source.network
* loki.source.push_api
* loki.source.receiver`